enterprise/backend/src/metabase_enterprise/gsheets/api.clj

```diff
@@ -180,10 +180,21 @@
     (throw (ex-info "Cannot fetch Google Drive connection: ID is nil" {})))
   (hm.client/make-request :get (str "/api/v2/mb/connections/" id)))
 
+(defn- url-type [url]
+  (cond
+    (re-matches #"^(https|http)://docs.google.com/spreadsheets/.*" url)
+    "google_spreadsheet"
+
+    (re-matches #"^(https|http)://drive.google.com/drive/.*" url)
+    "gdrive"
+
+    :else
+    (throw (ex-info (tru "Invalid URL: {0}" url) {:status-code 400}))))
```

Comment:
we need to support more types of links. apparently the mobile app generates links in this format

`https://drive.google.com/file/.*`

see https://metaboat.slack.com/archives/C079ZS06ERJ/p1738593116421199

Reply:
Support for this was added in the backend code